tests/Backend/AMQPBackendDispatcherTest.php

```diff
@@ -87,6 +132,32 @@ public function testShouldPassExpectedOptionsToAmqpConnectionFactoryConstructor(
         ], AmqpConnectionFactoryStub::$config);
     }
 
+    public function testShouldSetExpectedDelayStrategyToAmqpConnectionFactoryOfInstanceDelayStrategyAware()
+    {
+        $expectedContext = $this->createMock(AmqpContext::class);
+
+        $dispatcher = new AMQPBackendDispatcher(
+            [
+                'host' => 'aHost',
+                'port' => 'aPort',
+                'user' => 'aUser',
+                'pass' => 'aPass',
+                'vhost' => 'aVhost',
+                'factory_class' => AmqpConnectionFactoryDelayStrategyAwareStub::class,
+                'delay_strategy_class' => AnyDelayStrategyStub::class,
+            ],
+            [],
+            'default',
+            []
+        );
+
+        AmqpConnectionFactoryDelayStrategyAwareStub::$context = $expectedContext;
+
+        $actualContext = $dispatcher->getContext();
```

Comment:
Let's get rid of this one-time-variable, shall we?

Reply:
Do you think it should be removed from other tests too?

Reply:
Yes, in a separate commit and for bonus points :)